tests/test_flake8_trio.py

```diff
@@ -157,18 +158,30 @@ def check_autofix(
     assert added_autofix_diff == autofix_diff_content
 
 
-MAGIC_MARKERS = ("NOANYIO", "NOTRIO", "TRIO_NO_ERROR", "ANYIO_NO_ERROR")
+# This can be further cleaned up by adding the other return values from
+# parse_eval_file (Errors, args and enabled_codes) to this class - and find magic
+# markers in the same pass as we parse out errors etc.
+@dataclass
+class MagicMarkers:
+    NOANYIO: bool = False
+    NOTRIO: bool = False
+    ANYIO_NO_ERROR: bool = False
+    TRIO_NO_ERROR: bool = False
 
 
 def find_magic_markers(
     content: str,
-) -> dict[Literal["NOANYIO", "NOTRIO", "TRIO_NO_ERROR", "ANYIO_NO_ERROR"], bool]:
-    found_markers: dict[str, bool] = {m: False for m in MAGIC_MARKERS}
-    for f in re.findall(rf'# ({"|".join(MAGIC_MARKERS)})', content):
-        found_markers[f] = True
-    return found_markers  # type: ignore
+) -> MagicMarkers:
+    found_markers = MagicMarkers()
+    markers = (f.name for f in fields(found_markers))
+    pattern = rf'# ({"|".join(markers)})'
+    for f in re.findall(pattern, content):
+        setattr(found_markers, f, True)
+    return found_markers
 
 
+# This could be optimized not to reopen+reread+reparse the same file over and over
+# when testing the same file
```

Comment:
Turn `find_magic_markers()` into a classmethod on the dataclass, with `@lru_cache`?

Reply:
it's not just find_magic_markers, but also `_parse_eval_file`. The latter would need some additional logic since the errors & error messages will be slightly different when substituting trio for anyio.